src/common/hwio_buddy_2209_02.cpp

```diff
@@ -49,6 +49,9 @@ enum {
 };
 static const int _FAN_CNT = _FAN_ID_MAX - _FAN_ID_MIN + 1;
 
+metric_t metric_nozzle_PWM = METRIC("nozzle_PWM", METRIC_VALUE_INTEGER, 1000, METRIC_HANDLER_DISABLE_ALL);
+metric_t metric_bed_PWM = METRIC("bed_PWM", METRIC_VALUE_INTEGER, 1000, METRIC_HANDLER_DISABLE_ALL);
```

Comment:
/subjective, low priority/ I would slightly prefer `nozzle_pwn` and `bed_pwm` names. The original naming style was snake_case (lowercase only). However, it has been broken with some metrics already :(

Reply:
/suggestion/ This seem to record integers between 0-255. Please change to float values (0.0 - 1.0). The 8 bit values are implementation specific and we don't really care about them in metrics.

Reply:
> /suggestion/ This seem to record integers between 0-255. Please change to float values (0.0 - 1.0). The 8 bit values are implementation specific and we don't really care about them in metrics.

I don't like that suggestion. As using this raw value already helped me to find a bug - maximum value was 254 and not 255 as intended. Here we are implementing [Arduino analogWrite](https://www.arduino.cc/reference/en/language/functions/analog-io/analogwrite/) which has range 0 to 255 so it is kind a platform independent. Value in float is here not available - this would mean we need to compute it and I don't like that idea.

Reply:
> /subjective, low priority/ I would slightly prefer `nozzle_pwn` and `bed_pwm` names. The original naming style was snake_case (lowercase only). However, it has been broken with some metrics already :(

I can rename it if you wish.

Reply:
preferably using nozzle_pwm to nozzle_pwned

Reply:
I put the metric to Arduino layer, maybe it would make sense to put it even deeper and log value put to HW register - at this moment value range is the same 0 - 255 but it may change in future and it may help us to find potential error in abstraction layer.

Downside would be, that repeatedly setting the same value would not produce any metric. I don't know if it is good or bad.
We can even implement not-sending-the-same value right now for free if this is considered beneficial. In such case it would be worth to enable this metric during compile time forever, as it would not send useless data when printer is idling - PWM value is 0 then and would be sent just once.

Reply:
> > /suggestion/ This seem to record integers between 0-255. Please change to float values (0.0 - 1.0). The 8 bit values are implementation specific and we don't really care about them in metrics.
> 
> I don't like that suggestion. As using this raw value already helped me to find a bug - maximum value was 254 and not 255 as intended. Here we are implementing [Arduino analogWrite](https://www.arduino.cc/reference/en/language/functions/analog-io/analogwrite/) which has range 0 to 255 so it is kind a platform independent. Value in float is here not available - this would mean we need to compute it and I don't like that idea.

You would find the bug either way (as you wouldn't see the value 1.0, but 0.996). However I am fine with keeping it as a raw value...

Reply:
> Downside would be, that repeatedly setting the same value would not produce any metric. I don't know if it is good or bad. We can even implement not-sending-the-same value right now for free if this is considered beneficial. In such case it would be worth to enable this metric during compile time forever, as it would not send useless data when printer is idling - PWM value is 0 then and would be sent just once.

That is a wanted behaviour. We want to periodically send values for metrics even if they did not change.

Reply:
ok. Please mark it as resolved.